packages/generate-bot/test/generate-bot.js

```diff
@@ -0,0 +1,145 @@
+    }
+  });
+  console.log(contentString);
+  return contentString;
+};
+
+describe("file structure", () => {
+  it("checks that file structure carries over", async () => {
+    const originalStack = await recursive("./templates");
+    GenerateBot.creatingBotFiles("./templates", {
+      programName: "{{programName}}",
+      description: "description",
+      fileLocation: "./tmp"
+    });
+    let createdStack = await recursive("./tmp");
+    createdStack = createdStack.map(contents => {
+      return contents.replace(/tmp/, "templates");
+    });
+    console.log("OG " + originalStack);
+    console.log("CS " + createdStack);
+    expect(originalStack).to.have.members(createdStack);
+  });
+
+  afterEach(() => {
+    rimraf.sync("./tmp");
+  });
+
+  it("checks that the file content carries over", async () => {
+    GenerateBot.creatingBotFiles("./templates", {
+      programName: "helloWorld",
+      description: "says hi",
+      fileLocation: "./helloWorld"
+    });
+
+    const contentString = "Start of snapshot: ";
+    return snapshot(readAllFiles("./helloWorld", contentString));
+  });
+
+  afterEach(() => {
+    rimraf.sync("./helloWorld");
+  });
+});
+
+describe("user input", () => {
+  it("checks that user input is being checked correctly", () => {
+    let validityWorkingWell = true;
+    const hyphenTest = GenerateBot.checkValidity({
+      programName: "does-not-pass",
+      description: "pass",
+      fileLocation: "pass"
+    });
+
+    const integerTest = GenerateBot.checkValidity({
+      programName: "pass",
+      description: "5oesN0tP4SS",
+      fileLocation: "pass"
+    });
+
+    const nullTest = GenerateBot.checkValidity({
+      programName: "",
+      description: "pass",
+      fileLocation: "../pass"
+    });
+
+    const overWritingTest = GenerateBot.checkValidity({
+      programName: "templates",
+      description: "pass",
+      fileLocation: "./templates"
+    });
+
+    const programNameToLower = {
+      programName: "PassButMakeLowerCase",
+      description: "pass",
+      fileLocation: "pass"
+    };
+    GenerateBot.checkValidity(programNameToLower);
+
+    const fileLocationDefault = {
+      programName: "pass",
+      description: "pass",
+      fileLocation: ""
+    };
+    GenerateBot.checkValidity(fileLocationDefault);
+
+    console.log(hyphenTest);
+    console.log(nullTest);
+    console.log(integerTest);
+    console.log(overWritingTest);
+    console.log(programNameToLower.programName);
+    console.log(fileLocationDefault.fileLocation);
+
+    const regex = new RegExp("packages/pass");
+
+    if (
+      !hyphenTest &&
+      !nullTest &&
+      !integerTest &&
+      !overWritingTest &&
+      programNameToLower.programName === "passbutmakelowercase" &&
+      regex.test(fileLocationDefault.fileLocation)
+    ) {
+      validityWorkingWell = true;
+    } else {
+      validityWorkingWell = false;
+    }
+    expect(validityWorkingWell).to.be.true;
```

Comment:
something is breaking in windows, but it's hard to tell what exactly, because of all our assertions being combined, I would do this as:

```js
expect(nullTest).to.be.true;
expect(integerTest).to.be.true;
... etc.
```